serde.go

```diff
@@ -0,0 +1,43 @@
+package kafka
+
+import (
+	"errors"
+)
+
+type Serializer func(configuration Configuration, topic string, data interface{}, keyOrValue string, schema string) ([]byte, error)
+
+func GetSerializer(serializer string) Serializer {
+	switch serializer {
+	case "org.apache.kafka.common.serialization.ByteArraySerializer":
+		return SerializeByteArray
+	case "org.apache.kafka.common.serialization.StringSerializer":
+		return SerializeString
+	case "io.confluent.kafka.serializers.KafkaAvroSerializer":
+		return SerializeAvro
+	default:
+		return SerializeAvro
```

Comment:
I think the default case should be a string.

Reply:
I can change it to string, just thought to keep it compatible with some already existing tests which expect the previous behavior. Shall I still replace it?

Reply:
If e.g. Protobuf schemas will be supported in the future as well, then we should make it somehow required to specify the serializer not just providing a schema. Maybe the `produce` function should be dedicated to string serialization and the schema options should be removed. If a schema is used it should be provided using the `produceWithConfiguration` together with the appropriate serializer/deserializer configuration.

Reply:
The `produceWithConfiguration` isn't exported to JS because it's not an exported function in Go. Still, having a single uniform interface, `produce`, would be better if we could have optional arguments so that we don't pass an empty string for the schema (and possibly no nulls for others).
I made an initial proposal and built it on top of @iamelevich's work (https://github.com/mostafa/xk6-kafka/pull/20) to pass an object to the writer and the reader interface for configuration, but it didn't turn out as expected. I also included many other options which were left out in the writer configuration in the struct but to no avail.